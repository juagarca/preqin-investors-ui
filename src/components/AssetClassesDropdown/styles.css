.investors-dropdown select {
  width: 100%;
  padding: 0.5rem;
  margin-bottom: 1.5rem;
}
