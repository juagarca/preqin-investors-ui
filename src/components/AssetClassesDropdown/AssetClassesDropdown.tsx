import { ChangeEventHandler } from "react";

import { assetClasses } from "../../consts";
import { AssetClass } from "../../types";

import "./styles.css";

interface AssetClassesDropdownProps {
  assetClassSelected: AssetClass | undefined;
  handleChange: ChangeEventHandler<HTMLSelectElement>;
}

const AssetClassesDropdown = ({
  assetClassSelected,
  handleChange,
}: AssetClassesDropdownProps) => {
  return (
    <div className="investors-dropdown">
      <select id="dropdown" value={assetClassSelected} onChange={handleChange}>
        <option value="">
          Pick an asset class to show its commitment information
        </option>
        {assetClasses.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </div>
  );
};

export default AssetClassesDropdown;
