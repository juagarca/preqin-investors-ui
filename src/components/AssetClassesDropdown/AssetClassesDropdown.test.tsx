import React from "react";
import { render, screen } from "@testing-library/react";

import AssetClassesDropdown from "./AssetClassesDropdown";

describe("AssetClassesDropdown component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <AssetClassesDropdown
        assetClassSelected={undefined}
        handleChange={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with initial option", () => {
    render(
      <AssetClassesDropdown
        assetClassSelected={undefined}
        handleChange={() => {}}
      />
    );

    expect(screen.getByText(/Pick an asset class/)).toBeInTheDocument();
  });
});
