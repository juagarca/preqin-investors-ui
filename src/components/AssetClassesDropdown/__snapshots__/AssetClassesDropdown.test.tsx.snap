// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetClassesDropdown component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="investors-dropdown"
  >
    <select
      id="dropdown"
    >
      <option
        value=""
      >
        Pick an asset class to show its commitment information
      </option>
      <option
        value="pe"
      >
        Private Equity
      </option>
      <option
        value="pd"
      >
        Private Debt
      </option>
      <option
        value="re"
      >
        Real Estate
      </option>
      <option
        value="inf"
      >
        Infrastructure
      </option>
      <option
        value="nr"
      >
        Natural Resources
      </option>
      <option
        value="hf"
      >
        Hedge Funds
      </option>
    </select>
  </div>
</DocumentFragment>
`;
